packages/svelte/tests/runtime-runes/samples/effect-cleanup/main.svelte

```diff
@@ -0,0 +1,17 @@
+<script>
+	const {log} = $props();
+
+	let count = $state(0);
+
+	$effect(() => {
+		let double = $derived(count * 2)
+
+		log.push('init ' + double);
```

Comment:
I think this test can move into `tests/signals` - or do we need something Svelte-render-specific here? Seems purely related to the signals runtime.

Reply:
It matches the other tests inside the directory that checks logging order? It seems related to Svelte runtime too.

Reply:
Which other tests are you talking about? Maybe those other tests need to move into `tests/signals`, too 😄

Reply:
I tried, but I now get test failures accessing `component`. Let's leave them for now and address in another PR.

Reply:
what test failures about accessing `component`? When you try to turn this test into a test in `signals` then it fails or what do you mean exactly?

Reply:
The test suite setup is different I guess?

Reply:
So I've done some digging and ultimately I don't think these are signals tests and they shouldn't be coupled with being so. Signal tests should be agnostic of Svelte's components. However, these cases come from bug reports where users have created components that demonstrate ordering issues or timing issues that are relevant to their components. This is important as the signals tests also demonstrate unowned signals – which mean they are independent of components too, not to mention how this all fits into equality.

Let's merge this PR and create a separate issue on how we can unify the semantics around signals - vs - component state semantics, as I think there needs to be a distinction and having one is important. For example, our current signals tests are all wrapped with a render effect, which is likely wrong too. They shouldn't be, signals should be agnostic of effects.

Reply:
The idea of wrapping them is to simulate this being run inside a component context, which is why I was so confused why it wouldn't work when having them in that test suite